site/profile/manifests/core/anaconda.pp

```diff
@@ -0,0 +1,78 @@
+# @summary
+#   Installs anaconda
+#
+# @param python_env_name
+#   if exist, sets the name of the virtual environment
+#
+# @param python_env_version
+#   if exist, sets the version of python for the virtual environment
+#
+# @param conda_packages
+#   if contains at least 1 tuple (name and channel), will install the package
+#    
+class profile::core::anaconda (
+  String $python_env_name,
+  String $python_env_version,
+  Tuple   $conda_packages,
+) {
+  case $facts['os']['family'] {
+    'RedHat': {
+      ensure_packages('wget')
+      exec { 'install_anaconda':
+        command => '/bin/wget https://repo.anaconda.com/archive/Anaconda3-2023.07-2-Linux-x86_64.sh -O /tmp/anaconda_installer.sh && /bin/bash /tmp/anaconda_installer.sh -b -p /opt/anaconda',
+        creates => '/opt/anaconda/bin/conda',
+        path    => '/usr/bin:/bin:/opt/anaconda/bin',
+        require => Package['wget'],
+      }
+    }
+    default: {
+      notify { 'unsupported_os':
+        message => 'This Anaconda installation is only for RedHat family',
+      }
+    }
+  }
+  # Create a Conda environment with Python if parameters are set
+  if ($python_env_name and $python_env_version) {
+    exec { 'create_conda_env':
+      command => "/opt/anaconda/bin/conda create -y --name ${python_env_name} python=${python_env_version}",
+      creates => "/opt/anaconda/envs/${python_env_name}/",
+      require => Exec['install_anaconda'],
+    }
+    file { '/etc/profile.d/conda_source.sh':
+      ensure  => file,
+      mode    => '0644',
+      require => Exec['create_conda_env'],
+      # lint:ignore:strict_indent
+      content => @("SOURCE"),
+          #!/usr/bin/bash
```

Comment:
The shebang isn't needed as it is ignored when a file is sourced.

Reply:
the shebang here is out 👍🏻